### PR TITLE
dataapigeneratorjson/templater_terraform_mappings - Add additional ModelToModelMappings

### LINKED ISSUE #3267: Data API v2 bug: missing `ModelToModel` mapping for `KubernetesClusterTrustedAccessRoleBinding`
Part of putting together #3266 involved validating the output between Data API v1 and Data API v2 for mappings - as a part of that I've noticed that there's a missing `ModelToModel` mapping.

Data API v1 contains the following `modelToModel` mapping at the endpoint `http://localhost:5000/v1/resource-manager/services/ContainerService/terraform` for the `KubernetesClusterTrustedAccessRoleBinding` resource:

```json
        "modelToModel": [
          {
            "schemaModelName": "KubernetesClusterTrustedAccessRoleBindingResourceSchema",
            "sdkModelName": "TrustedAccessRoleBindingProperties"
          },
          {
            "schemaModelName": "KubernetesClusterTrustedAccessRoleBindingResourceSchema",
            "sdkModelName": "TrustedAccessRoleBinding"
          }
        ],
```

<details>
  <summary>Full Response from Data API v1:</summary>

```json
{
  "dataSources": {},
  "resources": {
    "kubernetes_cluster_trusted_access_role_binding": {
      "apiVersion": "2023-03-02-preview",
      "createMethod": {
        "generate": true,
        "methodName": "RoleBindingsCreateOrUpdate",
        "timeoutInMinutes": 30
      },
      "deleteMethod": {
        "generate": true,
        "methodName": "RoleBindingsDelete",
        "timeoutInMinutes": 30
      },
      "displayName": "Kubernetes Cluster Trusted Access Role Binding",
      "documentation": {
        "category": "Container",
        "description": "Manages a Kubernetes Cluster Trusted Access Role Binding\n~> **Note:** This Resource is in **Preview** to use this you must be opted into the Preview. You can do this by running \u0060az feature register --namespace Microsoft.ContainerService --name TrustedAccessPreview\u0060 and then \u0060az provider register -n Microsoft.ContainerService\u0060\n",
        "exampleUsageHcl": "resource \u0022azurerm_application_insights\u0022 \u0022example\u0022 {\n  name                = \u0022example\u0022\n  location            = azurerm_resource_group.example.location\n  resource_group_name = azurerm_resource_group.example.name\n  application_type    = \u0022example-value\u0022\n}\ndata \u0022azurerm_client_config\u0022 \u0022test\u0022 {}\nresource \u0022azurerm_key_vault\u0022 \u0022example\u0022 {\n  name                       = \u0022example\u0022\n  location                   = azurerm_resource_group.example.location\n  resource_group_name        = azurerm_resource_group.example.name\n  tenant_id                  = data.azurerm_client_config.example.tenant_id\n  sku_name                   = \u0022example-value\u0022\n  soft_delete_retention_days = \u0022example-value\u0022\n}\nresource \u0022azurerm_key_vault_access_policy\u0022 \u0022example\u0022 {\n  key_vault_id = azurerm_key_vault.example.id\n  tenant_id    = data.azurerm_client_config.example.tenant_id\n  object_id    = data.azurerm_client_config.example.object_id\n  key_permissions = \u0022example-value\u0022\n}\nresource \u0022azurerm_kubernetes_cluster\u0022 \u0022example\u0022 {\n  name                = \u0022example\u0022\n  location            = azurerm_resource_group.example.location\n  resource_group_name = azurerm_resource_group.example.name\n  dns_prefix          = \u0022acctestaksexample\u0022\n  default_node_pool {\n    name       = \u0022example-value\u0022\n    node_count = \u0022example-value\u0022\n    vm_size    = \u0022example-value\u0022\n  }\n  identity {\n    type = \u0022example-value\u0022\n  }\n}\nresource \u0022azurerm_machine_learning_workspace\u0022 \u0022example\u0022 {\n  name                    = \u0022example\u0022\n  location                = azurerm_resource_group.example.location\n  resource_group_name     = azurerm_resource_group.example.name\n  key_vault_id            = azurerm_key_vault.example.id\n  storage_account_id      = azurerm_storage_account.example.id\n  application_insights_id = azurerm_application_insights.example.id\n  identity {\n    type = \u0022example-value\u0022\n  }\n}\nresource \u0022azurerm_resource_group\u0022 \u0022example\u0022 {\n  name     = \u0022example-resources\u0022\n  location = \u0022West Europe\u0022\n}\nresource \u0022azurerm_storage_account\u0022 \u0022example\u0022 {\n  name                     = \u0022example\u0022\n  location                 = azurerm_resource_group.example.location\n  resource_group_name      = azurerm_resource_group.example.name\n  account_tier             = \u0022example-value\u0022\n  account_replication_type = \u0022example-value\u0022\n}\nresource \u0022azurerm_kubernetes_cluster_trusted_access_role_binding\u0022 \u0022example\u0022 {\n  kubernetes_cluster_id = azurerm_kubernetes_cluster.example.id\n  name                  = \u0022example\u0022\n  roles                 = \u0022example-value\u0022\n  source_resource_id    = azurerm_machine_learning_workspace.example.id\n}"
      },
      "generate": true,
      "generateModel": true,
      "generateIdValidation": true,
      "generateSchema": true,
      "mappings": {
        "fields": [
          {
            "type": "DirectAssignment",
            "directAssignment": {
              "schemaModelName": "KubernetesClusterTrustedAccessRoleBindingResourceSchema",
              "schemaFieldPath": "Roles",
              "sdkModelName": "TrustedAccessRoleBindingProperties",
              "sdkFieldPath": "Roles"
            },
            "modelToModel": null
          },
          {
            "type": "DirectAssignment",
            "directAssignment": {
              "schemaModelName": "KubernetesClusterTrustedAccessRoleBindingResourceSchema",
              "schemaFieldPath": "SourceResourceId",
              "sdkModelName": "TrustedAccessRoleBindingProperties",
              "sdkFieldPath": "SourceResourceId"
            },
            "modelToModel": null
          },
          {
            "type": "ModelToModel",
            "directAssignment": null,
            "modelToModel": {
              "schemaModelName": "KubernetesClusterTrustedAccessRoleBindingResourceSchema",
              "sdkModelName": "TrustedAccessRoleBinding",
              "sdkFieldName": "Properties"
            }
          }
        ],
        "modelToModel": [
          {
            "schemaModelName": "KubernetesClusterTrustedAccessRoleBindingResourceSchema",
            "sdkModelName": "TrustedAccessRoleBindingProperties"
          },
          {
            "schemaModelName": "KubernetesClusterTrustedAccessRoleBindingResourceSchema",
            "sdkModelName": "TrustedAccessRoleBinding"
          }
        ],
        "resourceId": [
          {
            "schemaFieldName": "KubernetesClusterId",
            "segmentName": "managedClusterName",
            "parsedFromParentId": true
          },
          {
            "schemaFieldName": "KubernetesClusterId",
            "segmentName": "resourceGroupName",
            "parsedFromParentId": true
          },
          {
            "schemaFieldName": "Name",
            "segmentName": "trustedAccessRoleBindingName",
            "parsedFromParentId": false
          }
        ]
      },
      "readMethod": {
        "generate": true,
        "methodName": "RoleBindingsGet",
        "timeoutInMinutes": 5
      },
      "resource": "TrustedAccess",
      "resourceIdName": "TrustedAccessRoleBindingId",
      "resourceName": "KubernetesClusterTrustedAccessRoleBinding",
      "schemaModelName": "KubernetesClusterTrustedAccessRoleBindingResourceSchema",
      "schemaModels": {
        "KubernetesClusterTrustedAccessRoleBindingResourceSchema": {
          "fields": {
            "KubernetesClusterId": {
              "objectDefinition": {
                "nestedObject": null,
                "referenceName": null,
                "type": "String"
              },
              "computed": false,
              "forceNew": true,
              "hclName": "kubernetes_cluster_id",
              "optional": false,
              "required": true,
              "documentation": {
                "markdown": "Specifies the Kubernetes Cluster Id within which this Kubernetes Cluster Trusted Access Role Binding should exist."
              },
              "validation": null
            },
            "Name": {
              "objectDefinition": {
                "nestedObject": null,
                "referenceName": null,
                "type": "String"
              },
              "computed": false,
              "forceNew": true,
              "hclName": "name",
              "optional": false,
              "required": true,
              "documentation": {
                "markdown": "Specifies the name of this Kubernetes Cluster Trusted Access Role Binding."
              },
              "validation": null
            },
            "Roles": {
              "objectDefinition": {
                "nestedObject": {
                  "nestedObject": null,
                  "referenceName": null,
                  "type": "String"
                },
                "referenceName": null,
                "type": "List"
              },
              "computed": false,
              "forceNew": false,
              "hclName": "roles",
              "optional": false,
              "required": true,
              "documentation": {
                "markdown": "A list of roles to bind, each item is a resource type qualified role name."
              },
              "validation": null
            },
            "SourceResourceId": {
              "objectDefinition": {
                "nestedObject": null,
                "referenceName": null,
                "type": "String"
              },
              "computed": false,
              "forceNew": true,
              "hclName": "source_resource_id",
              "optional": false,
              "required": true,
              "documentation": {
                "markdown": "The ARM resource ID of source resource that trusted access is configured for."
              },
              "validation": null
            }
          }
        }
      },
      "tests": {
        "basicConfiguration": "\nprovider \u0022azurerm\u0022 {\n  features {}\n}\n\nresource \u0022azurerm_kubernetes_cluster_trusted_access_role_binding\u0022 \u0022test\u0022 {\n  kubernetes_cluster_id = azurerm_kubernetes_cluster.test.id\n  name                  = \u0022acctestkctarb-${var.random_string}\u0022\n  roles                 = [\u0022Microsoft.MachineLearningServices/workspaces/mlworkload\u0022]\n  source_resource_id    = azurerm_machine_learning_workspace.test.id\n}\n    ",
        "requiresImportConfiguration": "\nresource \u0022azurerm_kubernetes_cluster_trusted_access_role_binding\u0022 \u0022import\u0022 {\n  kubernetes_cluster_id = azurerm_kubernetes_cluster_trusted_access_role_binding.test.kubernetes_cluster_id\n  name                  = azurerm_kubernetes_cluster_trusted_access_role_binding.test.name\n  roles                 = azurerm_kubernetes_cluster_trusted_access_role_binding.test.roles\n  source_resource_id    = azurerm_kubernetes_cluster_trusted_access_role_binding.test.source_resource_id\n}\n    ",
        "completeConfiguration": "\nprovider \u0022azurerm\u0022 {\n  features {}\n}\n\nresource \u0022azurerm_kubernetes_cluster_trusted_access_role_binding\u0022 \u0022test\u0022 {\n  kubernetes_cluster_id = azurerm_kubernetes_cluster.test.id\n  name                  = \u0022acctestkctarb-${var.random_string}\u0022\n  roles                 = [\u0022Microsoft.MachineLearningServices/workspaces/mlworkload\u0022, \u0022Microsoft.MachineLearningServices/workspaces/inference-v1\u0022]\n  source_resource_id    = azurerm_machine_learning_workspace.test.id\n}\n",
        "generate": true,
        "otherTests": {},
        "templateConfiguration": "\nvariable \u0022primary_location\u0022 {}\nvariable \u0022random_integer\u0022 {}\nvariable \u0022random_string\u0022 {}\n\nresource \u0022azurerm_application_insights\u0022 \u0022test\u0022 {\n  name                = \u0022acctestai-${var.random_integer}\u0022\n  location            = azurerm_resource_group.test.location\n  resource_group_name = azurerm_resource_group.test.name\n  application_type    = \u0022web\u0022\n}\n\n\ndata \u0022azurerm_client_config\u0022 \u0022test\u0022 {}\n\n\nresource \u0022azurerm_key_vault\u0022 \u0022test\u0022 {\n  name                       = \u0022acctest-${var.random_string}\u0022\n  location                   = azurerm_resource_group.test.location\n  resource_group_name        = azurerm_resource_group.test.name\n  tenant_id                  = data.azurerm_client_config.test.tenant_id\n  sku_name                   = \u0022standard\u0022\n  soft_delete_retention_days = 7\n}\n\n\nresource \u0022azurerm_key_vault_access_policy\u0022 \u0022test\u0022 {\n  key_vault_id = azurerm_key_vault.test.id\n  tenant_id    = data.azurerm_client_config.test.tenant_id\n  object_id    = data.azurerm_client_config.test.object_id\n\n  key_permissions = [\n    \u0022Create\u0022,\n    \u0022Get\u0022,\n    \u0022Delete\u0022,\n    \u0022Purge\u0022,\n    \u0022GetRotationPolicy\u0022,\n  ]\n}\n\n\nresource \u0022azurerm_kubernetes_cluster\u0022 \u0022test\u0022 {\n  name                = \u0022acctestaks${var.random_string}\u0022\n  location            = azurerm_resource_group.test.location\n  resource_group_name = azurerm_resource_group.test.name\n  dns_prefix          = \u0022acctestaks${var.random_string}\u0022\n\n  default_node_pool {\n    name       = \u0022default\u0022\n    node_count = 1\n    vm_size    = \u0022Standard_DS2_v2\u0022\n  }\n\n  identity {\n    type = \u0022SystemAssigned\u0022\n  }\n}\n\n\nresource \u0022azurerm_machine_learning_workspace\u0022 \u0022test\u0022 {\n  name                    = \u0022acctestmlw-${var.random_integer}\u0022\n  location                = azurerm_resource_group.test.location\n  resource_group_name     = azurerm_resource_group.test.name\n  key_vault_id            = azurerm_key_vault.test.id\n  storage_account_id      = azurerm_storage_account.test.id\n  application_insights_id = azurerm_application_insights.test.id\n\n  identity {\n    type = \u0022SystemAssigned\u0022\n  }\n}\n\n\nresource \u0022azurerm_resource_group\u0022 \u0022test\u0022 {\n  name     = \u0022acctestrg-${var.random_integer}\u0022\n  location = var.primary_location\n}\n\n\nresource \u0022azurerm_storage_account\u0022 \u0022test\u0022 {\n  name                     = \u0022acctestsa${var.random_string}\u0022\n  location                 = azurerm_resource_group.test.location\n  resource_group_name      = azurerm_resource_group.test.name\n  account_tier             = \u0022Standard\u0022\n  account_replication_type = \u0022LRS\u0022\n}\n"
      },
      "updateMethod": {
        "generate": true,
        "methodName": "RoleBindingsCreateOrUpdate",
        "timeoutInMinutes": 30
      }
    },
    "kubernetes_fleet_manager": {
      "apiVersion": "2022-09-02-preview",
      "createMethod": {
        "generate": true,
        "methodName": "CreateOrUpdate",
        "timeoutInMinutes": 30
      },
      "deleteMethod": {
        "generate": true,
        "methodName": "Delete",
        "timeoutInMinutes": 30
      },
      "displayName": "Kubernetes Fleet Manager",
      "documentation": {
        "category": "Container",
        "description": "Manages a Kubernetes Fleet Manager\n\n~> **Note:** This Resource is in **Preview** to use this you must be opted into the Preview. You can do this by running \u0060az feature register --namespace Microsoft.ContainerService --name FleetResourcePreview\u0060 and then \u0060az provider register -n Microsoft.ContainerService\u0060\n",
        "exampleUsageHcl": "resource \u0022azurerm_resource_group\u0022 \u0022example\u0022 {\n  name     = \u0022example-resources\u0022\n  location = \u0022West Europe\u0022\n}\nresource \u0022azurerm_kubernetes_fleet_manager\u0022 \u0022example\u0022 {\n  location            = azurerm_resource_group.example.location\n  name                = \u0022example\u0022\n  resource_group_name = azurerm_resource_group.example.name\n}"
      },
      "generate": true,
      "generateModel": true,
      "generateIdValidation": true,
      "generateSchema": true,
      "mappings": {
        "fields": [
          {
            "type": "DirectAssignment",
            "directAssignment": {
              "schemaModelName": "KubernetesFleetManagerResourceFleetHubProfileSchema",
              "schemaFieldPath": "DnsPrefix",
              "sdkModelName": "FleetHubProfile",
              "sdkFieldPath": "DnsPrefix"
            },
            "modelToModel": null
          },
          {
            "type": "DirectAssignment",
            "directAssignment": {
              "schemaModelName": "KubernetesFleetManagerResourceFleetHubProfileSchema",
              "schemaFieldPath": "Fqdn",
              "sdkModelName": "FleetHubProfile",
              "sdkFieldPath": "Fqdn"
            },
            "modelToModel": null
          },
          {
            "type": "DirectAssignment",
            "directAssignment": {
              "schemaModelName": "KubernetesFleetManagerResourceFleetHubProfileSchema",
              "schemaFieldPath": "KubernetesVersion",
              "sdkModelName": "FleetHubProfile",
              "sdkFieldPath": "KubernetesVersion"
            },
            "modelToModel": null
          },
          {
            "type": "DirectAssignment",
            "directAssignment": {
              "schemaModelName": "KubernetesFleetManagerResourceSchema",
              "schemaFieldPath": "HubProfile",
              "sdkModelName": "FleetProperties",
              "sdkFieldPath": "HubProfile"
            },
            "modelToModel": null
          },
          {
            "type": "DirectAssignment",
            "directAssignment": {
              "schemaModelName": "KubernetesFleetManagerResourceSchema",
              "schemaFieldPath": "Location",
              "sdkModelName": "Fleet",
              "sdkFieldPath": "Location"
            },
            "modelToModel": null
          },
          {
            "type": "DirectAssignment",
            "directAssignment": {
              "schemaModelName": "KubernetesFleetManagerResourceSchema",
              "schemaFieldPath": "Tags",
              "sdkModelName": "Fleet",
              "sdkFieldPath": "Tags"
            },
            "modelToModel": null
          },
          {
            "type": "ModelToModel",
            "directAssignment": null,
            "modelToModel": {
              "schemaModelName": "KubernetesFleetManagerResourceFleetHubProfileSchema",
              "sdkModelName": "FleetProperties",
              "sdkFieldName": "HubProfile"
            }
          },
          {
            "type": "ModelToModel",
            "directAssignment": null,
            "modelToModel": {
              "schemaModelName": "KubernetesFleetManagerResourceSchema",
              "sdkModelName": "Fleet",
              "sdkFieldName": "Properties"
            }
          }
        ],
        "modelToModel": [
          {
            "schemaModelName": "KubernetesFleetManagerResourceFleetHubProfileSchema",
            "sdkModelName": "FleetHubProfile"
          },
          {
            "schemaModelName": "KubernetesFleetManagerResourceSchema",
            "sdkModelName": "FleetProperties"
          },
          {
            "schemaModelName": "KubernetesFleetManagerResourceSchema",
            "sdkModelName": "Fleet"
          },
          {
            "schemaModelName": "KubernetesFleetManagerResourceFleetHubProfileSchema",
            "sdkModelName": "FleetProperties"
          }
        ],
        "resourceId": [
          {
            "schemaFieldName": "Name",
            "segmentName": "fleetName",
            "parsedFromParentId": false
          },
          {
            "schemaFieldName": "ResourceGroupName",
            "segmentName": "resourceGroupName",
            "parsedFromParentId": false
          }
        ]
      },
      "readMethod": {
        "generate": true,
        "methodName": "Get",
        "timeoutInMinutes": 5
      },
      "resource": "Fleets",
      "resourceIdName": "FleetId",
      "resourceName": "KubernetesFleetManager",
      "schemaModelName": "KubernetesFleetManagerResourceSchema",
      "schemaModels": {
        "KubernetesFleetManagerResourceFleetHubProfileSchema": {
          "fields": {
            "DnsPrefix": {
              "objectDefinition": {
                "nestedObject": null,
                "referenceName": null,
                "type": "String"
              },
              "computed": false,
              "forceNew": false,
              "hclName": "dns_prefix",
              "optional": false,
              "required": true,
              "documentation": {
                "markdown": null
              },
              "validation": null
            },
            "Fqdn": {
              "objectDefinition": {
                "nestedObject": null,
                "referenceName": null,
                "type": "String"
              },
              "computed": true,
              "forceNew": false,
              "hclName": "fqdn",
              "optional": false,
              "required": false,
              "documentation": {
                "markdown": null
              },
              "validation": null
            },
            "KubernetesVersion": {
              "objectDefinition": {
                "nestedObject": null,
                "referenceName": null,
                "type": "String"
              },
              "computed": true,
              "forceNew": false,
              "hclName": "kubernetes_version",
              "optional": false,
              "required": false,
              "documentation": {
                "markdown": null
              },
              "validation": null
            }
          }
        },
        "KubernetesFleetManagerResourceSchema": {
          "fields": {
            "HubProfile": {
              "objectDefinition": {
                "nestedObject": null,
                "referenceName": "KubernetesFleetManagerResourceFleetHubProfileSchema",
                "type": "Reference"
              },
              "computed": false,
              "forceNew": true,
              "hclName": "hub_profile",
              "optional": true,
              "required": false,
              "documentation": {
                "markdown": "The FleetHubProfile configures the Fleet\u0027s hub."
              },
              "validation": null
            },
            "Location": {
              "objectDefinition": {
                "nestedObject": null,
                "referenceName": null,
                "type": "Location"
              },
              "computed": false,
              "forceNew": true,
              "hclName": "location",
              "optional": false,
              "required": true,
              "documentation": {
                "markdown": "The Azure Region where the Kubernetes Fleet Manager should exist."
              },
              "validation": null
            },
            "Name": {
              "objectDefinition": {
                "nestedObject": null,
                "referenceName": null,
                "type": "String"
              },
              "computed": false,
              "forceNew": true,
              "hclName": "name",
              "optional": false,
              "required": true,
              "documentation": {
                "markdown": "Specifies the name of this Kubernetes Fleet Manager."
              },
              "validation": null
            },
            "ResourceGroupName": {
              "objectDefinition": {
                "nestedObject": null,
                "referenceName": null,
                "type": "ResourceGroup"
              },
              "computed": false,
              "forceNew": true,
              "hclName": "resource_group_name",
              "optional": false,
              "required": true,
              "documentation": {
                "markdown": "Specifies the name of the Resource Group within which this Kubernetes Fleet Manager should exist."
              },
              "validation": null
            },
            "Tags": {
              "objectDefinition": {
                "nestedObject": null,
                "referenceName": null,
                "type": "Tags"
              },
              "computed": false,
              "forceNew": false,
              "hclName": "tags",
              "optional": true,
              "required": false,
              "documentation": {
                "markdown": "A mapping of tags which should be assigned to the Kubernetes Fleet Manager."
              },
              "validation": null
            }
          }
        }
      },
      "tests": {
        "basicConfiguration": "\nprovider \u0022azurerm\u0022 {\n  features {}\n}\n\nresource \u0022azurerm_kubernetes_fleet_manager\u0022 \u0022test\u0022 {\n  location            = azurerm_resource_group.test.location\n  name                = \u0022acctestkfm-${var.random_string}\u0022\n  resource_group_name = azurerm_resource_group.test.name\n}\n    ",
        "requiresImportConfiguration": "\nresource \u0022azurerm_kubernetes_fleet_manager\u0022 \u0022import\u0022 {\n  location            = azurerm_kubernetes_fleet_manager.test.location\n  name                = azurerm_kubernetes_fleet_manager.test.name\n  resource_group_name = azurerm_kubernetes_fleet_manager.test.resource_group_name\n}\n    ",
        "completeConfiguration": "\nprovider \u0022azurerm\u0022 {\n  features {}\n}\n\nresource \u0022azurerm_kubernetes_fleet_manager\u0022 \u0022test\u0022 {\n  location            = azurerm_resource_group.test.location\n  name                = \u0022acctestkfm-${var.random_string}\u0022\n  resource_group_name = azurerm_resource_group.test.name\n  tags = {\n    environment = \u0022terraform-acctests\u0022\n    some_key    = \u0022some-value\u0022\n  }\n  hub_profile {\n    dns_prefix = \u0022val-${var.random_string}\u0022\n  }\n}\n",
        "generate": true,
        "otherTests": {},
        "templateConfiguration": "\nvariable \u0022primary_location\u0022 {}\nvariable \u0022random_integer\u0022 {}\nvariable \u0022random_string\u0022 {}\n\nresource \u0022azurerm_resource_group\u0022 \u0022test\u0022 {\n  name     = \u0022acctestrg-${var.random_integer}\u0022\n  location = var.primary_location\n}\n"
      },
      "updateMethod": {
        "generate": true,
        "methodName": "CreateOrUpdate",
        "timeoutInMinutes": 30
      }
    }
  }
}
```

</details>

The Data API v1 mappings for this resource are:

```csharp
namespace Pandora.Definitions.ResourceManager.ContainerService.Terraform;

public class KubernetesClusterTrustedAccessRoleBindingResourceMappings : TerraformMappingDefinition
{
    public List<MappingType> Mappings => new List<MappingType>
    {
        Mapping.FromSchema<KubernetesClusterTrustedAccessRoleBindingResourceSchema>(s => s.KubernetesClusterId).ToCommonIdSegmentNamed("managedClusterName"),
        Mapping.FromSchema<KubernetesClusterTrustedAccessRoleBindingResourceSchema>(s => s.KubernetesClusterId).ToCommonIdSegmentNamed("resourceGroupName"),
        Mapping.FromSchema<KubernetesClusterTrustedAccessRoleBindingResourceSchema>(s => s.Name).ToResourceIdSegmentNamed("trustedAccessRoleBindingName"),

        Mapping.FromSchema<KubernetesClusterTrustedAccessRoleBindingResourceSchema>(s => s.Roles).ToSdkField<TrustedAccessRoleBindingPropertiesModel>(m => m.Roles).Direct(),
        Mapping.FromSchema<KubernetesClusterTrustedAccessRoleBindingResourceSchema>(s => s.SourceResourceId).ToSdkField<TrustedAccessRoleBindingPropertiesModel>(m => m.SourceResourceId).Direct(),
        Mapping.FromSchemaModel<KubernetesClusterTrustedAccessRoleBindingResourceSchema>().ToSdkField<TrustedAccessRoleBindingModel>(m => m.Properties).ModelToModel(),
    };
}
```

The V2 API Definitions however contain:

```json
{
	"fieldMappings": [
		{
			"type": "DirectAssignment",
			"directAssignment": {
				"schemaModelName": "KubernetesClusterTrustedAccessRoleBindingResource",
				"schemaFieldPath": "Roles",
				"sdkModelName": "TrustedAccessRoleBindingProperties",
				"sdkFieldPath": "Roles"
			}
		},
		{
			"type": "DirectAssignment",
			"directAssignment": {
				"schemaModelName": "KubernetesClusterTrustedAccessRoleBindingResource",
				"schemaFieldPath": "SourceResourceId",
				"sdkModelName": "TrustedAccessRoleBindingProperties",
				"sdkFieldPath": "SourceResourceId"
			}
		},
		{
			"type": "ModelToModel",
			"modelToModel": {
				"schemaModelName": "KubernetesClusterTrustedAccessRoleBindingResource",
				"sdkModelName": "TrustedAccessRoleBinding",
				"sdkFieldName": "Properties"
			}
		}
	],
	"modelToModelMappings": [
		{
			"schemaModelName": "KubernetesClusterTrustedAccessRoleBindingResource",
			"sdkModelName": "TrustedAccessRoleBinding"
		}
	],
	"resourceIdMappings": [
		{
			"parsedFromParentId": true,
			"schemaFieldName": "KubernetesClusterId",
			"segmentName": "managedClusterName"
		},
		{
			"parsedFromParentId": true,
			"schemaFieldName": "KubernetesClusterId",
			"segmentName": "managedClusterName"
		},
		{
			"parsedFromParentId": false,
			"schemaFieldName": "Name",
			"segmentName": "trustedAccessRoleBindingName"
		}
	]
}
```

### PR DESCRIPTION
**Pull request by mbfrahry**
Fixes #3267

Also fixes a bug where we weren't removing the duplicated ModelToModelMappings

### CODE REVIEW

File: tools/importer-rest-api-specs/components/dataapigeneratorjson/templater_terraform_mappings.go [hunk 1]
```diff
@@ -170,19 +175,15 @@ func orderFieldMappings(input []dataApiModels.TerraformFieldMappingDefinition) (
...
 func uniqueAndSortModelToModelMappings(input []dataApiModels.TerraformModelToModelMappingDefinition) []dataApiModels.TerraformModelToModelMappingDefinition {
-	keys := make([]string, 0)
 	keysToValues := make(map[string]dataApiModels.TerraformModelToModelMappingDefinition)
 	for _, item := range input {
 		// using this key format means we'll also de-dupe this slice at the same time
 		key := fmt.Sprintf("%s-%s", item.SchemaModelName, item.SdkModelName)
-		keys = append(keys, key)
 		keysToValues[key] = item
 	}
-	sort.Strings(keys)
```
Comment (tombuildsstuff) [inline tools/importer-rest-api-specs/components/dataapigeneratorjson/templater_terraform_mappings.go:181]: This sort is used to ensure the output is consistent, meaning that we don't get spurious changes every regeneration - so this is likely still needed?

Instead we should be able to build up the `keys` map in a separate for loop, to give us a unique, ordered list of keys to map the values across for below

**Review by tombuildsstuff — CHANGES_REQUESTED** [review]
One comment about ordering, but this otherwise LGTM 👍

File: tools/importer-rest-api-specs/components/dataapigeneratorjson/templater_terraform_mappings.go (hunk 1 above)
Comment (mbfrahry) [reply tools/importer-rest-api-specs/components/dataapigeneratorjson/templater_terraform_mappings.go:181]: Good shout! Done!

**Review by stephybun — APPROVED** [review]
Thanks @mbfrahry LGTM 🧂

**Pull request merged**